### ISSUE TITLE
GraphQL::Schema::Mutation resolver can't return "type"

### ISSUE THREAD

**Issue by mech** [rmosolgo/graphql-ruby#3963]
**Describe the bug**

My app is working on v2.0.0, but break under v.2.0.1. I am using GraphQL::Schema::Mutation with a custom return Types::BaseObject, however it keep on wanting to return the auto-generated return type of XXXPayload.

**Versions**

`graphql` version: 2.0.1
`rails` (or other framework): 7.0.2.2

**GraphQL schema**

```ruby
class CompanyCreate < GraphQL::Schema::Mutation
  argument :input, Client::CompanyInput, required: true

  type MutationResponse, null: false

  def resolve(input:)
  end 
end

class MutationResponse < Types::BaseObject
  field :errors, [ActiveModelError], null: false
  field :item, ResponseItem, null: true

  def errors
    object.fetch(:errors, [])
  end

  def item
    object.fetch(:item, nil)
  end
end
```

Note that my return type is `MutationResponse`, however, v2.0.1 want to autogenerate `CompanyCreatePayload` instead. The above code does work on v2.0.0 though.

**Comment by rmosolgo** [comment]
Thanks for the report, and sorry for the trouble! I'll work up a fix soon and follow up here.

**Comment by rmosolgo** [comment]
I just released a patch in graphql-ruby 2.0.2. Thanks again for letting me know about this -- and please let me know if you encounter any more trouble with this after updating!
